R/pmcmc_parameters.R

```diff
@@ -30,9 +34,13 @@
 ##' @examples
 ##' pmcmc_parameter("a", 0.1)
 pmcmc_parameter <- function(name, initial, min = -Inf, max = Inf,
-                            discrete = FALSE, prior = NULL) {
+                            discrete, integer = FALSE, prior = NULL) {
+  if (!missing(discrete)) {
```

Comment:
This really highlights that there's some general parameter duplication we need to tidy up, probably before we get the ode stuff merged

Reply:
Same with the checking logic; the min/max/discrete checks need to run in a few places, so should be some way to share that code, whether a base parameter class (is that a thing in R?) or a shared function.

Reply:
Shared function probably easiest. You can do base classes with R6 but they're not used here